gptme/config.py

```diff
@@ -205,16 +164,144 @@ def get_project_config(workspace: Path | None) -> ProjectConfig | None:
         )
         # load project config
         with open(project_config_path) as f:
-            config_data = dict(tomlkit.load(f))
+            config_data = tomlkit.load(f)
 
         # Handle RAG config conversion before creating ProjectConfig
         if "rag" in config_data:
             config_data["rag"] = RagConfig(**config_data["rag"])  # type: ignore
 
-        return ProjectConfig(**config_data)  # type: ignore
+        mcp = config_data.pop("mcp", {})
+        mcp_config = MCPConfig.from_toml(mcp)
+
+        return ProjectConfig(**config_data, mcp=mcp_config)  # type: ignore
     return None
 
 
+@dataclass
+class ChatConfig:
```

Comment:
ChatConfig should maybe have a `from_logpath` which takes the path to a conversation directory (with `chat.toml`, `conversation.jsonl`, `workspace` dir/symlink, etc., `branches` dir) and loads the chat config if it exists. Then perhaps create a function on top which handles the case where we aren't loading but creating a new conversation.

Reply:
Done.